python/interpret-core/interpret/glassbox/_ebm/_utils.py

```diff
@@ -32,11 +57,11 @@ def convert_categorical_to_continuous(categories):
     # We can't convert a continuous feature that has cuts back into categoricals
     # since the categorical value could have been anything between the cuts that we know about.
 
-    clusters = dict()
+    clusters = defaultdict(list)
     non_float_idxs = set()
 
-    old_min = np.nan
-    old_max = np.nan
+    old_min = +np.inf
+    old_max = -np.inf
```

Comment:
Using +inf and -inf changes the semantics of the function slightly.  If none of the categories are finite numbers (which would hit the continue statements in the loop below) then this function would return np.nan for the min and max, while here it would return min=+inf and max=-inf

Reply:
However, looking at the function a bit longer, I see that I had a bug below in the check "if len(clusters) <= 1" which would subsequently return a single empty array instead of returning "cuts, mapping, old_min, old_max" as is done at the bottom, so the +inf and -inf are not an issue if that return is changed to return np.nan values for the min and max

Reply:
If you'd like to fix my bug below please feel free, otherwise I'll correct it after merging this PR.  I think your code above doesn't need any changes if this bug is fixed.

Reply:
You are welcome to fix this, as you can probably also add a test-case covering these special cases. I still haven't completely figured out want kind of inputs this function will process.

Reply:
Ok, sounds good!